tests/unit/test_endpoints.py

```diff
@@ -294,3 +296,33 @@ def __init__(self, service, old_endpoints, new_endpoints):
 
     def run(self):
         assert_equal(self.old_endpoints, self.new_endpoints)
+
+
+def test_no_lost_services():
+    # This makes sure that a bad sync doesn't cause us to loose any services
+    supported_services = [
+        'autoscaling', 'awslambda', 'cloudformation', 'cloudfront', 'cloudhsm',
+        'cloudsearch', 'cloudtrail', 'cloudwatch', 'codedeploy',
+        'cognito-identity', 'cognito-sync', 'configservice', 'datapipeline',
+        'directconnect', 'dynamodb', 'ec2', 'ec2containerservice',
+        'elasticache', 'elasticbeanstalk', 'elasticloadbalancing',
+        'elasticmapreduce', 'elastictranscoder', 'glacier', 'iam',
+        'importexport', 'kinesis', 'kms', 'logs', 'machinelearning',
+        'opsworks', 'rds', 'redshift', 'route53', 'route53domains', 's3',
+        'sdb', 'ses', 'sns', 'sqs', 'storagegateway', 'sts', 'support', 'swf'
+    ]
+
+    with open(boto.ENDPOINTS_PATH) as f:
+        endpoints = json.load(f)
+
+    resolver = BotoEndpointResolver(endpoints)
+    available_services = resolver.get_available_services()
+
+    for service in supported_services:
```

Comment:
We talked about expanding this to include regions and hostnames so the current endpoint in the data directory will not regress for users that did not opt in to it.